packages/core/src/components/axes/grid-brush.ts

```diff
@@ -49,13 +82,25 @@ export class ChartBrush extends Component {
 			this.frontSelectionSelector
 		);
 
+		const setDomain = (newDomain) => {
+			this.model.set({ zoomDomain: newDomain }, { animate: false });
+		};
+
+		const getDefaultZoomBarDomain = () =>
+			this.services.zoom.getDefaultZoomBarDomain();
+
 		if (mainXScale && mainXScaleType === ScaleTypes.TIME) {
 			// get current zoomDomain
 			let zoomDomain = this.model.get("zoomDomain");
 			if (zoomDomain === undefined) {
 				// default to full range with extended domain
-				zoomDomain = this.services.zoom.getDefaultZoomBarDomain();
-				this.model.set({ zoomDomain: zoomDomain }, { animate: false });
+				zoomDomain = getDefaultZoomBarDomain();
+				if (zoomDomain) {
+					this.model.set(
```

Comment:
why not use `setDomain` here?

Reply:
Good catch!
Fixed in https://github.com/carbon-design-system/carbon-charts/pull/717/commits/20627d79c5a281be09c6252b7e28bc9d05d0d013.